scripts/operations/gateway-test/cn-gateway-test.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+if [ $# -ne 1 ]; then
+  echo "usage: $0 <cn-hostname>"
+  exit 1
+fi
+
+CN_NAME=$1
+
+# Make sure we can resolve the CN Name
+if [ $(dig +short ${CN_NAME} | wc -l) -lt 1 ]; then
+    echo "ERROR: Unknown CN ${CN_NAME}"
+    exit 1
+fi
+
+# Make sure the cray, craysys, kubectl, and jq commands are available
+
+craysys type get > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+  echo "ERROR: craysys command is not available"
+  exit 1
+fi
+
+kubectl version > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+  echo "ERROR: kubectl command is not available"
+  exit 1
+fi
+
+which jq > /dev/null 2>&1
+if [ $? -ne 0 ]; then
+  echo "ERROR: jq command is not available"
+  exit 1
+fi
+
+# Get the Base Directory
+BASEDIR=$(dirname $0)
+
+# Get the SYSTEM_DOMAIN from cloud-init 
+SYSTEM_NAME=$(craysys metadata get system-name)
+SITE_DOMAIN=$(craysys metadata get site-domain)
+SYSTEM_DOMAIN=${SYSTEM_NAME}.${SITE_DOMAIN}
+echo "System domain is ${SYSTEM_DOMAIN}"
+
+# Get a token to talk to SLS
+export TOKEN=$(curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
```

Comment:
Should we validate that we got a token? It is true that if we don't get a token, then the subsequent curl command will fail. But it seems like it would make debug easier if it was clear that no token was obtained, rather than that the curl command failed for some other eason.

Reply:
Fixed